scripts/refcodes_with_properties/EntryPropertyCalculator.py

```diff
@@ -0,0 +1,582 @@
+
+class _ComparativeFilter(_Filter):
+    def __init__(self, args):
+        value = False
+        if eval(args.strip()) == 1:
+           value = True
+
+        self.expected_value = value
+
+    def value(self):
+        raise NotImplementedError # override this
+
+    def __call__(self,theobject):
+        value = self.value(theobject)
+        return value == self.expected_value
+
+
+class _RangeFilter(_Filter):
+    def __init__(self, args):
+
+        parts = [ p.strip() for p in args.split() ]
+        self.minimum = eval(parts[0])
+        self.maximum = eval(parts[1])
+
+    def value(self):
+        raise NotImplementedError # override this
+
+    def __call__(self,theobject):
+        value = self.value(theobject)
+        return value >= self.minimum and value <= self.maximum
+
+class AllowedAtomicNumbersFilter(_Filter):
+    def __init__(self,args):
+        self.allowed_atomic_numbers = [eval(x) for x in args.strip().split()]
+
+    @staticmethod
+    def name():
+        return "allowed atomic numbers"
+
+    @staticmethod
+    def helptext():
+        return "specify a set of atomic numbers (space separated) that the structure can have (and no others)"
+
+    def __call__(self,entry):
+        try:
+            molecule = entry.crystal.molecule
+            return len([x for x in molecule.atoms if x.atomic_number in self.allowed_atomic_numbers]) == len(molecule.atoms)
+        except TypeError:
+            return False
+
+register(AllowedAtomicNumbersFilter)
+
+
+class MustContainAtomicNumbersFilter(_Filter):
+    def __init__(self,args):
+        self.must_have_atomic_numbers = [eval(x) for x in args.strip().split()]
+
+    @staticmethod
+    def name():
+        return "must have atomic numbers"
+
+    @staticmethod
+    def helptext():
+        return "specify a set of atomic numbers (space separated) that the structure must have"
+
+    def __call__(self,entry):
+        try:
+            molecule = entry.crystal.molecule
+
+            contains = {}
+            for x in molecule.atoms:
+               if not contains.has_key(x.atomic_number):
+                   contains[x.atomic_number] = 0
+               contains[x.atomic_number] = contains[x.atomic_number] + 1
+            for x in self.must_have_atomic_numbers:
+                if not contains.has_key(x):
+                   return False
+
+            return True
+        except:
+            return False
+
+register(MustContainAtomicNumbersFilter)
+
+class OrganicFilter(_ComparativeFilter):
+    def __init__(self, args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "organic"
+
+    @staticmethod
+    def helptext():
+        return "organic entries or not"
+
+    def value(self,entry):
+        return entry.is_organic
+
+register(OrganicFilter)
+
+class PolymericFilter(_ComparativeFilter):
+    def __init__(self, args):
+        super(self.__class__,self).__init__(args)
```

Comment:
Same for the other supers

Reply:
I think this probably shows just how old this code is ...